### PR TITLE
Fixing regression in LDAP state module

### LINKED ISSUE #60760: Regression in state.ldap module
**Description**
```
----------
          ID: ldapi:///
    Function: ldap.managed
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python3/dist-packages/salt/state.py", line 2171, in call
                  ret = self.states[cdata["full"]](
                File "/usr/lib/python3/dist-packages/salt/loader.py", line 1241, in __call__
                  return self.loader.run(run_func, *args, **kwargs)
                File "/usr/lib/python3/dist-packages/salt/loader.py", line 2274, in run
                  return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
                File "/usr/lib/python3/dist-packages/salt/loader.py", line 2289, in _run_as
                  return _func_or_method(*args, **kwargs)
                File "/usr/lib/python3/dist-packages/salt/loader.py", line 2322, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python3/dist-packages/salt/states/ldap.py", line 257, in managed
                  old, new = _process_entries(l, entries)
                File "/usr/lib/python3/dist-packages/salt/states/ldap.py", line 450, in _process_entries
                  _update_entry(newe, entry_status, directives)
                File "/usr/lib/python3/dist-packages/salt/states/ldap.py", line 483, in _update_entry
                  vals.update(entry.get(attr, ()))
              AttributeError: 'list' object has no attribute 'update'
```

**Setup**
```
      - cn=Administrator:
        - default:
            cn:
              - Administrator
            objectClass:
              - organizationalRole
        - add:
            roleOccupant:
              - somecn
```
  
**Steps to Reproduce the behavior**
Use sls above and run `salt-call`

**Expected behavior**
salt not crashing

**Versions Report**
```
Salt Version:
          Salt: 3003.2

Dependency Versions:
          cffi: Not Installed
      cherrypy: Not Installed
      dateutil: 2.7.3
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 2.10.1
       libgit2: Not Installed
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 0.6.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: Not Installed
      pycrypto: Not Installed
  pycryptodome: 3.6.1
        pygit2: Not Installed
        Python: 3.8.10 (default, Jun  2 2021, 10:49:15)
  python-gnupg: 0.4.5
        PyYAML: 5.3.1
         PyZMQ: 18.1.1
         smmap: Not Installed
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.2

System Versions:
          dist: ubuntu 20.04 focal
        locale: utf-8
       machine: x86_64
       release: 5.4.0-81-generic
        system: Linux
       version: Ubuntu 20.04 focal

```
</details>

**Additional context**
I suppose that the issue is produced by ea30f817942c9c5ee59bbd19b1c0fbcee8102844

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Fixing regression in the ldap state module when adding attributes to an LDAP object.

### What issues does this PR fix or reference?
Fixes: #60760 

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: tests/unit/states/test_ldap.py [hunk 1]
```diff
@@ -8,13 +8,17 @@
...
 import copy
+import logging
 
 import salt.states.ldap
 from salt.utils.oset import OrderedSet
 from salt.utils.stringutils import to_bytes
 from tests.support.mixins import LoaderModuleMockMixin
 from tests.support.unit import TestCase
 
+log = logging.getLogger(__name__)
```
Comment (Ch3LL) [inline tests/unit/states/test_ldap.py:19]: I'm not seeing log being used anywhere.

Also, can these tests be converted to pytest?

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: tests/pytests/unit/states/test_ldap.py [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+import copy
+
+import pytest
+import salt.states.ldap
+from salt.utils.oset import OrderedSet
+from salt.utils.stringutils import to_bytes
+
+# emulates the LDAP database.  each key is the DN of an entry and it
+# maps to a dict which maps attribute names to sets of values.
+db = {}
```
Comment (Ch3LL) [inline tests/pytests/unit/states/test_ldap.py:19]: we want to avoid using a mutable object on the global/module level. Is this possible?

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: tests/pytests/unit/states/test_ldap.py (hunk 2 above)
Comment (garethgreenaway) [reply tests/pytests/unit/states/test_ldap.py:19]: Thought I had fixed that one, will take a look.

Comment (s0undt3ch) [reply tests/pytests/unit/states/test_ldap.py:19]: Make it a fixture.

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Review by Ch3LL — DISMISSED** [review]

**Pull request merged**